### DISCUSSION TITLE
create workspace: pull image: pull image: Cannot connect to the Docker daemon

### DISCUSSION THREAD

**Discussion by dotbenj** [coder/code-server#4820] [Q&A]
Hello there. I'm trying to test the coder-server on docker but when I tried to create workspaces I'm facing this issue:

`create workspace: pull image: pull image: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

So it seems the docker run for coder cannot access docker O_o
Does anybody had this issue and know how to fix this?

**Comment by lg-nineoneninethreenine** [comment]
Mostly, if you are testing on Docker, you might need to clone deploy-code-server. After cloning it, try building the Docker image. If it still fails, @bpmct might give more insight on this.

**Reply by dotbenj** [reply]
Hey thanks for your response. I've tried using the docker image of `deploy-code-server` on the docker repository as the doc said:
```
docker run --rm -it \
   -p 7080:7080 \
   -v /var/run/docker.sock:/var/run/docker.sock \
   -v ~/.coder:/var/run/coder \
   codercom/coder:1.27.0

```
I think the docker repo is up to date with github repo. What will be different if I'm building new image from from a clone of the github repo?

**Reply by jsjoeio** [reply]
Hmm...I'm not sure what this is or how to fix it. To be clear, are you running code-server like the [docs suggest](https://coder.com/docs/code-server/latest/install#docker), or are you trying Coder for Docker?

**Reply by lg-nineoneninethreenine** [reply]
It's Coder for Docker.

**Reply by jawnsy** [reply]
Here are a few ideas for debugging this:

1. Verify that docker is running (`ps aux | grep dockerd` on Linux systems)
2. Check that you have permissions to access the Docker socket (on some systems, you need to either be root or part of the `docker` group)
3. Check that `docker info` returns some output (if you get the same error, then Docker may not be running or you may not have permissions to use the socket to communicate with the daemon)
4. Verify that you can run other images (try `docker run debian`)

I'm not familiar with the behavior on MacOS or Windows, so if you're using those operating systems, you may need to search around or try Docker's troubleshooting documentation https://docs.docker.com/desktop/mac/troubleshoot/

**Reply by dotbenj** [reply]
> Hmm...I'm not sure what this is or how to fix it. To be clear, are you running code-server like the [docs suggest](https://coder.com/docs/code-server/latest/install#docker), or are you trying Coder for Docker?

Ho sorry I messed up its CODER and not code-server I'm trying to run with docker I probably an in the wrong repo to ask this question but thanks for your response anyway.

**Reply by dotbenj** [reply]
> Here are a few ideas for debugging this:
> 
> 1. Verify that docker is running (`ps aux | grep dockerd` on Linux systems)
> 2. Check that you have permissions to access the Docker socket (on some systems, you need to either be root or part of the `docker` group)
> 3. Check that `docker info` returns some output (if you get the same error, then Docker may not be running or you may not have permissions to use the socket to communicate with the daemon)
> 4. Verify that you can run other images (try `docker run debian`)
> 
> I'm not familiar with the behavior on MacOS or Windows, so if you're using those operating systems, you may need to search around or try Docker's troubleshooting documentation https://docs.docker.com/desktop/mac/troubleshoot/

I'll check that thanks a lot. FYI I'm running it on ubuntu.
